File: src/main/java/com/github/unchama/seichiassist/task/AsyncEntityRemover.java
package com.github.unchama.seichiassist.task;

import com.github.unchama.seichiassist.SeichiAssist;
import org.bukkit.entity.Entity;
import org.bukkit.scheduler.BukkitRunnable;

public class AsyncEntityRemover extends BukkitRunnable{
	Entity e;

	public AsyncEntityRemover(Entity e) {
		this.e = e;
	}

	@Override
	public void run() {
		SeichiAssist.entitylist().remove(e);
		e.remove();
	}

}

File: src/main/scala/com/github/unchama/seichiassist/task/AsyncEntityRemover.scala
package com.github.unchama.seichiassist.task

import com.github.unchama.seichiassist.SeichiAssist
import org.bukkit.entity.Entity
import org.bukkit.scheduler.BukkitRunnable

class AsyncEntityRemover(var e: Entity) extends BukkitRunnable {
  override def run(): Unit = {
    SeichiAssist.managedEntities.$minus$eq(e)
    e.remove()
  }
}